tests/test_utils/api_connexion_utils.py

```diff
@@ -30,6 +30,19 @@ def create_user(app, username, role):
         )
 
 
+def create_role(app, name, permissions=None):
```

Comment:
What do you think about extending the create_user method to accept a list of permissions?
```
def create_user(app, username, role=None, permissions=None):
    appbuilder = app.appbuilder
    current_role = None
    if role and permission:
    	raise IllegalArgument("role and permission are muttually exclussive")
    elif role:
	    current_role = appbuilder.sm.find_role(role)
	elif permission:
		ranadom_suffix = get_random_string()
	    current_role = create_role(app, get_random_string("test-role-{ranadom_suffix}"))

    tester = appbuilder.sm.find_user(username=username)
    if not tester:
        appbuilder.sm.add_user(
            username=username,
            first_name=username,
            last_name=username,
            email=f"{username}@fab.org",
            role=current_role,
            password=username,
        )
```
This will simplify testing as we won't have to worry so much about side effects as each role will only be used once. This will also explicitly associate the user with a specific permission set.